src/login/commands/selectTenant.ts

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { commands, window } from "vscode";
+import { callWithTelemetryAndErrorHandling, IActionContext } from "vscode-azureextensionui";
+import { tenantSetting } from "../../constants";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../utils/localize";
+import { openUri } from "../../utils/openUri";
+import { updateSettingValue } from "../../utils/settingUtils";
+import { AzureResourceFilterInternal } from "../subscriptionTypes";
+
+export async function selectTenant(): Promise<void> {
+	await callWithTelemetryAndErrorHandling('azure-account.selectTenant', async (context: IActionContext) => {
+		if (!(await ext.loginHelper.api.waitForFilters())) {
+			context.telemetry.properties.outcome = 'notLoggedIn';
+			return commands.executeCommand('azure-account.askForLogin');
+		}
+
+		const tenantSet = new Set<string>();
+		for (const filter of ext.loginHelper.api.filters) {
+			for (const tenant of (<AzureResourceFilterInternal>filter).tenants) {
+				tenantSet.add(tenant);
+			}
+		}
+		const tenants: string[] = Array.from(tenantSet);
+
+		if (!tenants.length) {
+			context.telemetry.properties.outcome = 'noTenantsFound';
+			const noTenantsFound = localize('azure-account.noTenantsFound', 'No tenants were found for the subscription(s) you\'ve selected.');
+			const learnMoreAboutTenants = { title: localize('azure-account.learnMoreAboutTenants', 'Learn more about tenants') };
+			void context.ui.showWarningMessage(noTenantsFound, learnMoreAboutTenants).then(result => {
+				if (result === learnMoreAboutTenants) {
+					void openUri('https://aka.ms/AAfe10l');
+				}
+			});
```

Comment:
I'm open to input as to whether or not this is needed... It seems like a nice addition but I'm not sure how often this case will happen. Also since we provide the option to enter a custom tenant the quickpick list will ever be empty.